core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4703,28 +4741,33 @@ private QueryModel rewriteSampleBy(@Nullable QueryModel model) throws SqlExcepti
 
                 int timestampPos = model.getColumnAliasIndex(timestampAlias);
 
-                // create function ast
-                final ExpressionNode timestampFunc = expressionNodePool.next();
+                ExpressionNode timestampFunc = expressionNodePool.next();
                 timestampFunc.token = "timestamp_floor";
-                timestampFunc.paramCount = 2;
                 timestampFunc.type = FUNCTION;
+                timestampFunc.paramCount = sampleByFrom != null ? 3 : 2;
 
                 CharacterStoreEntry characterStoreEntry = characterStore.newEntry();
                 characterStoreEntry.put('\'').put(sampleBy.token).put('\'');
 
-                final ExpressionNode lhs = expressionNodePool.next();
-                lhs.token = characterStoreEntry.toImmutable();
-                lhs.paramCount = 0;
-                lhs.type = CONSTANT;
-
-                final ExpressionNode rhs = expressionNodePool.next();
-                rhs.token = timestampColumn;
-                rhs.position = timestamp.position;
-                rhs.paramCount = 0;
-                rhs.type = LITERAL;
-
-                timestampFunc.lhs = lhs;
-                timestampFunc.rhs = rhs;
+                final ExpressionNode param1 = expressionNodePool.next();
+                param1.token = characterStoreEntry.toImmutable();
+                param1.paramCount = 0;
+                param1.type = CONSTANT;
+
+                final ExpressionNode param2 = expressionNodePool.next();
+                param2.token = timestampColumn;
+                param2.position = timestamp.position;
+                param2.paramCount = 0;
+                param2.type = LITERAL;
+
+                if (sampleByFrom != null) {
```

Comment:
nit: would be nice to put a clarifying comment here to make the intent more obvious for new readers